torch/_C/_distributed_c10d.pyi

```diff
@@ -134,8 +134,9 @@ class ReduceOptions:
     rootTensor: int
     timeout: timedelta
 
-class AllGatherOptions:
```

Comment:
curious: why the rename here? (reduce-scatter is still using ReduceScatterOptions)

Reply:
Yeah, the original name seems to be a typo. And it never caused a problem because this option was never imported before.
In `init.cpp`, the class we defined has a little `g`:
```
py::class_<::c10d::AllgatherOptions>(module, "AllgatherOptions")
```
so I am fixing the name in the pyi file to stay the same.